DrinkManagerWeb/Controllers/AdminController.cs

```diff
@@ -0,0 +1,327 @@
+﻿using BLL;
+using DrinkManagerWeb.Models.ViewModels;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace DrinkManagerWeb.Controllers
+{
+    [Authorize(Roles = "Admin")]
+    public class AdminController : Controller
+    {
+        private readonly RoleManager<IdentityRole> _roleManager;
+        private readonly UserManager<AppUser> _userManager;
+        private readonly IPasswordHasher<AppUser> _passwordHasher;
+
+        public AdminController(RoleManager<IdentityRole> roleManager, UserManager<AppUser> userManager, IPasswordHasher<AppUser> passwordHasher)
+        {
+            _roleManager = roleManager;
+            _userManager = userManager;
+            _passwordHasher = passwordHasher;
+        }
+
+        public IActionResult Index()
+        {
+            return RedirectToAction("Users");
+        }
+
+        public IActionResult Users()
+        {
+            var users = _userManager.Users.OrderBy(user => user.UserName).ToList();
+            var usersAndRoles = new Dictionary<string, List<string>>();
+
+            foreach (var user in users)
+            {
+                var roles = _userManager.GetRolesAsync(user).Result.ToList();
+                usersAndRoles.Add(user.UserName, roles);
+            }
+
+            var model = new UserListViewModel { RolesPerUser = usersAndRoles, Users = users };
+
+            return View(model);
+        }
+
+        public ViewResult CreateUser()
+        {
+            var model = new UserViewModel
+            {
+                ApplicationRoles = _roleManager.Roles
+                    .Select(r => new SelectListItem { Text = r.Name, Value = r.Id })
+                    .ToList()
+            };
+            return View(model);
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> CreateUser(UserViewModel model)
+        {
+            if (ModelState.IsValid)
+            {
+                var role = await _roleManager.FindByIdAsync(model.ApplicationRoleId);
+                if (role != null)
+                {
+                    AppUser user = new AppUser
+                    {
+                        Email = model.Email,
+                        UserName = model.Email
+                    };
+
+                    IdentityResult result = await _userManager.CreateAsync(user, model.Password);
+
+                    if (result.Succeeded)
+                    {
+                        await _userManager.AddToRoleAsync(user, role.Name);
+                        return RedirectToAction("Users");
+                    }
```

Comment:
if createAsync result has errors we should add them to the model state

                foreach (var error in result.Errors)
                {
                    ModelState.AddModelError(string.Empty, error.Description);
                }

Reply:
Done.